requirements-dev.txt

```diff
@@ -42,7 +42,7 @@ websockets>=9.1
 psutil
 
 # For running the python-checker tests
-hacking==6.1.0
-flake8==6.1.0
-pylint==3.0.3
-mypy==1.8.0
+hacking>=6.1.0
+flake8>=6.1.0
+pylint>=3.0.3
+mypy>=1.8.0
```

Comment:
I guess you realize what this means (different people with different versions of these tools will get different linter warnings), so no need for me to point this out.

Besides, as things now stand, it is already true that sometimes the os versions and sometimes the pypi venv versions of these tools are used, depending on how one sets up their dev environment, so this is not actually something entirely new. What's now new is that this will be also seen in CI (new ci failures on python liner checks because pypi packages were updated, without any code changes in router)

Reply:
Yeah - this change is a bit heavy handed. I had to advance pylint (I think it was pylint) as the 3.0.3 version was actually failing on F40 (some type assertion).

Having newer linter versions catch new errors is a good thing... probably. If it introduces lots of noise then yeah we should set upper version bounds as necessary.